packages/solidity-contracts/protocol/utils.ts

```diff
@@ -49,3 +52,90 @@ export function tai64Time(millis: number): string {
     .add(zeroPointOffset)
     .toHexString();
 }
+
+// Computes data for message
+export function computeMessageData(
+  fuelContractId: string,
+  tokenAddress: string,
+  tokenId: BigNumberish,
+  from: string,
+  to: string,
+  amount: number,
+  data?: BytesLike
+): string {
+  if (data) {
+    const depositToContractFlag = ethers.utils
+      .keccak256(ethers.utils.toUtf8Bytes('DEPOSIT_TO_CONTRACT'))
+      .substring(0, 4);
+    if (data.length == 0) {
+      return ethers.utils.solidityPack(
+        [
+          'bytes32',
+          'bytes32',
+          'uint256',
+          'bytes32',
+          'bytes32',
+          'uint256',
+          'bytes1',
+        ],
+        [
+          fuelContractId,
+          tokenAddress,
+          BigNumber.from(tokenId),
+          from,
+          to,
+          amount,
+          depositToContractFlag,
+        ]
+      );
```

Comment:
For code quality (CQ) I would assign these two arrays to variables at the top. In the `if` expression I would check if the length is not 0. If the length is not 0 then I would append the `bytes` and `data` to the arrays.

At the end I would throw them into a singular return.

```typescript
let types = ["bytes32", ...];
let messageData = [fuelContractId, ...];

if (data.length > 0) {
    // append to arrays
}

return ethers.utils.solidityPack(types, messageData);
```

I see there's an extra return for when the data is not provided so the variables could be factored out to the top.

Reply:
Done, resolving